internal/server/config.go

```diff
@@ -33,6 +34,16 @@ type Jobs struct {
 	ExpiringAccessNotification JobConfig `mapstructure:"expiring_access_notification"`
 }
 
+type DefaultAuth struct {
+	HeaderKey string `mapstructure:"header_key" default:"X-Auth-Email"`
+}
+
+type Auth struct {
+	Provider string        `mapstructure:"provider" default:"default"`
+	Default  DefaultAuth   `mapstructure:"default"`
+	OIDC     auth.OIDCAuth `mapstructure:"oidc"`
+}
+
 type Config struct {
```

Comment:
let's also add `// Deprecated: use Auth.Default.HeaderKey instead` note on the `AuthenticatedUserHeaderKey` to mark the field as deprecated and suggest to use the new one